File: solutions/java/reverse-string/2/src/main/java/ReverseString.java
class ReverseString {

    String reverse(String inputString) {
        StringBuilder rev = new StringBuilder(); 
        for (int i = inputString.length() - 1; i >= 0; i--) {
            rev.append(inputString.charAt(i));
        }
        return rev.toString();
    }
}

File: solutions/c/reverse-string/1/reverse_string.c
#include "reverse_string.h"
#include <stdlib.h>

char *reverse(const char *value) {
    if (!value)
        return NULL;

    int len = 0;
    while(value[len])
        len++;

    char *ptr = malloc(len + 1);

    if (!ptr)
        return NULL;

    for (int i = 0; i < len; i++)
        ptr[i] = value[len - i - 1];

    ptr[len] = '\0';

    return ptr;
}